### PR TITLE
KFSPTS-8359: Add personal expense handling to SAE-to-Collector process.

### PR DESCRIPTION
**Pull request by cah292**
This is a corrected version of the feature from PR #386 that was later reverted. There are only a couple minor changes in this PR compared to the old one, mostly revolving around what account number to use for some of the new generated offsets.

This code is ready to be merged once the unit tests pass and the PR assignees are satisfied with it.

### CODE REVIEW

**Comment by cah292** [comment]
I've removed @db483 from this PR since she will be unavailable to review it on Monday. If others think another assignee needs to be added in her place, please let me know.

File: src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java [hunk 1]
```diff
@@ -205,6 +240,90 @@ protected KualiDecimal calculateTotalAmountForLines(List<ConcurStandardAccountin
...
+            if (totalSubGroupAmount.isNonZero()) {
+                ConcurStandardAccountingExtractDetailLine firstCreditLine = corpCardPersonalCredits.get(0);
+                addOriginEntryForCorpCardPersonalExpense(
+                        entryConsumer, firstCreditLine, collectorHelper.getPrepaidOffsetAccountNumber(), StringUtils.EMPTY,
+                        collectorHelper.getPrepaidOffsetObjectCode(), totalSubGroupAmount);
+            }
+        }
+    }
+
+    protected void addOriginEntriesForCorpCardPersonalExpenseDebitLines(Consumer<OriginEntryFull> entryConsumer) {
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java]: This is a pretty long function,  can it be refactored into some smaller functions?

Comment (cah292) [inline src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java]: Ok, I'll work on refactoring this accordingly.

**Comment by cah292** [comment]
I've made some updates based on feedback. This is now ready for review again.

**Comment by cah292** [comment]
retest this please

**Pull request merged**